pkg/validation-webhook/snapshot.go

```diff
@@ -223,6 +254,43 @@ func decideSnapshotContentV1(snapcontent, oldSnapcontent *volumesnapshotv1.Volum
 	return reviewResponse
 }
 
+func decideSnapshotClassV1(snapClass, oldSnapClass *volumesnapshotv1.VolumeSnapshotClass, lister storagelisters.VolumeSnapshotClassLister) *v1.AdmissionResponse {
+	reviewResponse := &v1.AdmissionResponse{
+		Allowed: true,
+		Result:  &metav1.Status{},
+	}
+
+	// Only Validate when a new snapClass is being set as a default.
+	if snapClass.Annotations[utils.IsDefaultSnapshotClassAnnotation] != "true" {
+		return reviewResponse
+	}
+
+	// If Old snapshot class has this, then we can assume that it was validated
+	if oldSnapClass.Annotations[utils.IsDefaultSnapshotClassAnnotation] == "true" {
+		return reviewResponse
+	}
```

Comment:
This is tricky. As I pointed out above, if `driver:` is changed in the new VSC, it may be the second default VSC for the new driver and some validation is needed.

IMO it would make sense not to allow change of `driver:` in any VSC, we don't allow it in StorageClasses too. Then the `if` above will be correct.

Or re-check classes of the new driver if driver name is changed.

Reply:
I will add this as a test case and fix it. I should have thought of this sorry.

Reply:
I think we probably don't want to change existing behavior right?

Reply:
> I think we probably don't want to change existing behavior right?

It's tempting, but no, let's allow changing VSC.driver.